includes/plugins/class-newspack-newsletters.php

```diff
@@ -119,179 +117,6 @@ public static function get_metadata_key( $key ) {
 	}
 
-	/**
-	 * Update content metadata after a contact's lists are updated.
-	 *
-	 * @param string        $provider        The provider name.
-	 * @param string        $email           Contact email address.
-	 * @param string[]      $lists_to_add    Array of list IDs to subscribe the contact to.
-	 * @param string[]      $lists_to_remove Array of list IDs to remove the contact from.
-	 * @param bool|WP_Error $result          True if the contact was updated or error if failed.
-	 */
-	public static function update_contact_lists( $provider, $email, $lists_to_add, $lists_to_remove, $result ) {
-		switch ( $provider ) {
-			case 'active_campaign':
-				if ( true === $result && method_exists( '\Newspack_Newsletters_Subscription', 'add_contact' ) && method_exists( '\Newspack_Newsletters_Subscription', 'get_contact_lists' ) ) {
-					$current_lists = \Newspack_Newsletters_Subscription::get_contact_lists( $email );
-					// The add_contact method is idempotent, effectively being an upsertion.
-					\Newspack_Newsletters_Subscription::add_contact( [ 'email' => $email ], $current_lists );
-				}
-				break;
-		}
-	}
-
-	/**
-	 * Modify metadata for newsletter contact creation.
-	 *
-	 * @param array          $contact           {
-	 *          Contact information.
-	 *
-	 *    @type string   $email                 Contact email address.
-	 *    @type string   $name                  Contact name. Optional.
-	 *    @type string   $existing_contact_data Existing contact data, if updating a contact. The hook will be also called when
-	 *    @type string[] $metadata              Contact additional metadata. Optional.
-	 * }
-	 * @param string[]|false $selected_list_ids Array of list IDs the contact will be subscribed to, or false.
-	 * @param string         $provider          The provider name.
-	 */
-	public static function contact_data( $contact, $selected_list_ids, $provider ) {
-		switch ( $provider ) {
-			case 'active_campaign':
-				$metadata = [];
-				if ( is_user_logged_in() ) {
-					$metadata[ self::get_metadata_key( 'account' ) ] = get_current_user_id();
-				}
-
-				// Translate list IDs to list names and store as metadata, if lists are supplied.
-				// The list ids can be an empty array, which means the contact has been unsubscribed from all lists.
-				if ( false !== $selected_list_ids ) {
-					try {
-						if ( method_exists( '\Newspack_Newsletters_Subscription', 'get_lists' ) ) {
-							$lists = \Newspack_Newsletters_Subscription::get_lists();
-							if ( ! is_wp_error( $lists ) ) {
-								$lists_names = [];
-								foreach ( $selected_list_ids as $selected_list_id ) {
-									foreach ( $lists as $list ) {
-										if ( $list['id'] === $selected_list_id ) {
-											$lists_names[] = $list['name'];
-										}
-									}
-								}
-								// Note: this field will be overwritten every time it's updated.
-								$metadata[ self::get_metadata_key( 'newsletter_selection' ) ] = implode( ', ', $lists_names );
```

Comment:
I assume this metadata is deprecated? It makes sense because it duplicates what's already available in AC, but I wanted to make sure.

Reply:
Thank you for catching that. Although I agree it's duplicated data, I don't think we should deprecate it without a broader conversation.

Implemented support for the key in 42ea81210b9f891bd2fc1349a09e9e3645682a10

You should be able to see the `Newsletter Selection` key updating in the contact on the `newsletter_updated` data event. This happens when subscribing or changing the newsletter selection in "My Account -> Newsletters".

Reply:
> This happens when subscribing or changing the newsletter selection in "My Account -> Newsletters".

It should also happen whenever someone subscribes. When testing, I subscribe via the Newsletter Subscribe block, and the metadata is not updated then.

Reply:
Added in 26f93d2f83b04168680a6b75645a996e34e2520e